Sources/Datadog/RUM/RUMTelemetry.swift

```diff
@@ -18,43 +18,34 @@ internal final class RUMTelemetry: Telemetry {
     static let MaxEventsPerSessions: Int = 100
 
     let core: DatadogCoreProtocol
-    let sdkVersion: String
-    let applicationID: String
-    let source: String
     let dateProvider: DateProvider
-    let dateCorrector: DateCorrector
     let sampler: Sampler
 
     /// Keeps track of current session
-    private var currentSessionID: RUMUUID = .nullUUID
+    private var currentSessionID: String?
 
     /// Keeps track of event's ids recorded during a user session.
     private var eventIDs: Set<String> = []
 
+    /// Queue for processing RUM Telemetry
+    private let queue = DispatchQueue(
+        label: "com.datadoghq.rum-telemetry",
+        target: .global(qos: .utility)
+    )
```

Comment:
`RUMTelemetry` no longer depend on the global `RUMMonitor` and its queue. This new queue sync and serialise telemetry updates.